patches/chromium/convert_electron_processsingleton_changes_to_patch.patch

```diff
@@ -0,0 +1,264 @@
+     const base::CommandLine& command_line,
+@@ -997,12 +1030,26 @@ bool ProcessSingleton::Create() {
+ #endif
+   }
+ 
+-  // Create the socket file somewhere in /tmp which is usually mounted as a
+-  // normal filesystem. Some network filesystems (notably AFS) are screwy and
+-  // do not support Unix domain sockets.
+-  if (!socket_dir_.CreateUniqueTempDir()) {
+-    LOG(ERROR) << "Failed to create socket directory.";
+-    return false;
++  if (IsAppSandboxed()) {
++    // For sandboxed applications, the tmp dir could be too long to fit
++    // addr->sun_path, so we need to make it as short as possible.
++    base::FilePath tmp_dir;
++    if (!base::GetTempDir(&tmp_dir)) {
++      LOG(ERROR) << "Failed to get temporary directory.";
++      return false;
++    }
++    if (!socket_dir_.Set(tmp_dir.Append("S"))) {
++      LOG(ERROR) << "Failed to set socket directory.";
++      return false;
++    }
++  } else {
++    // Create the socket file somewhere in /tmp which is usually mounted as a
++    // normal filesystem. Some network filesystems (notably AFS) are screwy and
++    // do not support Unix domain sockets.
++    if (!socket_dir_.CreateUniqueTempDir()) {
++      LOG(ERROR) << "Failed to create socket directory.";
++      return false;
++    }
+   }
+ 
+   // Check that the directory was created with the correct permissions.
+@@ -1044,10 +1091,13 @@ bool ProcessSingleton::Create() {
+   if (listen(sock, 5) < 0)
+     NOTREACHED() << "listen failed: " << base::safe_strerror(errno);
+ 
+-  DCHECK(BrowserThread::IsThreadInitialized(BrowserThread::IO));
+-  content::GetIOThreadTaskRunner({})->PostTask(
+-      FROM_HERE, base::BindOnce(&ProcessSingleton::LinuxWatcher::StartListening,
+-                                watcher_, sock));
++  sock_ = sock;
++
++  if (BrowserThread::IsThreadInitialized(BrowserThread::IO)) {
++    StartListeningOnSocket();
++  } else {
++    listen_on_ready_ = true;
++  }
+ 
+   return true;
+ }
+diff --git a/chrome/browser/process_singleton_win.cc b/chrome/browser/process_singleton_win.cc
+index f1732f042c3bfd9d5d95a4208664f9252b2aab73..49075cd6064e5189b96221aff51be1c5af176d18 100644
+--- a/chrome/browser/process_singleton_win.cc
++++ b/chrome/browser/process_singleton_win.cc
+@@ -27,7 +27,6 @@
+ #include "base/win/windows_version.h"
+ #include "base/win/wmi.h"
+ #include "chrome/browser/shell_integration.h"
+-#include "chrome/browser/ui/simple_message_box.h"
+ #include "chrome/browser/win/chrome_process_finder.h"
+ #include "chrome/common/chrome_constants.h"
+ #include "chrome/common/chrome_paths.h"
+@@ -174,11 +173,10 @@ bool ProcessLaunchNotification(
+   return true;
+ }
+ 
+-bool DisplayShouldKillMessageBox() {
+-  return chrome::ShowQuestionMessageBoxSync(
+-             NULL, l10n_util::GetStringUTF16(IDS_PRODUCT_NAME),
+-             l10n_util::GetStringUTF16(IDS_BROWSER_HUNGBROWSER_MESSAGE)) !=
+-         chrome::MESSAGE_BOX_RESULT_NO;
++bool TerminateAppWithError() {
```

Comment:
This part reverts back to the previous behaviour in `chromium_src`.
It also saves us from having to link a bunch more files specifically for https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/views/message_box_dialog.cc;l=319;drc=6f5405753a101d6059b45d702a3bcec75a28c114;bpv=1;bpt=1